modules/planning/navi/common/local_path.h

```diff
@@ -0,0 +1,109 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ * @brief This file provides the declaration of the class "LocalPath".
+ */
+
+#ifndef MODULES_PLANNING_NAVI_LOCAL_PATH_H_
+#define MODULES_PLANNING_NAVI_LOCAL_PATH_H_
+
+#include <vector>
+
+#include "modules/common/math/vec2d.h"
+#include "modules/common/proto/pnc_point.pb.h"
+
+/**
+ * @namespace apollo::planning
+ * @brief apollo::planning
+ */
+namespace apollo {
+namespace planning {
+
+using apollo::common::math::Vec2d;
+/**
+ * @class LocalPath
+ * @brief LocalPath is used to privide some path point operations for decider
+ */
+class LocalPath {
+ public:
+  LocalPath(std::vector<common::PathPoint> path_points)
+      : path_points_(path_points) {
+    points_.clear();
+    for (int i = 0; i < path_points_.size(); ++i) {
+      points_.emplace_back(path_points_[i].x(), path_points_[i].y());
+    }
+  }
+
+  virtual ~LocalPath() = default;
+
+  /**
+   * @brief get the first y of points vector.
+   * @param y input parm and the output.
+   * @return bool whether the y is valid or not.
+   */
+  bool GetInitY(double &y);
```

Comment:
use a pointer for output, unless there is a strong reason not to do so (such as swap()).